packages/backend/src/api/event/eventRouter.ts

```diff
@@ -0,0 +1,176 @@
+
+export const eventRouter: Router = (() => {
+  const router = express.Router();
+
+  eventRegistry.registerPath({
+    method: 'get',
+    path: '/events',
+    tags: ['Event'],
+    responses: createApiResponse(z.array(EventSchema), 'Success'),
+  });
+
+  router.get('/', validateRequest(GetAllEventSchema), async (req: Request, res: Response) => {
+    const { pageIndex, pageSize, typeId, title } = req.query as unknown as GetEventRequest;
+    const serviceResponse = await eventService.findAll({
+      pageIndex,
+      pageSize,
+      typeId,
+      title,
+    });
+    handleServiceResponse(serviceResponse, res);
+  });
+
+  eventRegistry.registerPath({
+    method: 'get',
+    path: '/events/{id}',
+    tags: ['Event'],
+    responses: createApiResponse(z.array(EventSchema), 'Success'),
+  });
+
+  router.get('/:id', validateRequest(GetEventSchema), async (req: Request, res: Response) => {
+    console.log('req.params', req.params);
+    const id = req.params.id as string;
+    const serviceResponse = await eventService.findById(id);
+    handleServiceResponse(serviceResponse, res);
+  });
+
+  eventRegistry.registerPath({
+    method: 'post',
+    path: '/events',
+    tags: ['Event'],
+    request: {
+      params: CreateEventSchema,
+    },
+    responses: createApiResponse(EventSchema, 'Success'),
+  });
+
+  router.post('/', async (req: Request, res: Response) => {
+    try {
+      const parsedEvent = CreateEventSchema.parse({ body: req.body });
+      const serviceResponse = await eventService.create(parsedEvent);
+      handleServiceResponse(serviceResponse, res);
+    } catch (error) {
+      if (error instanceof ZodError) {
+        const err = new ServiceResponse(ResponseStatus.Failed, error.message, null, StatusCodes.BAD_REQUEST);
+        return handleServiceResponse(err, res);
+      }
+    }
+  });
+
+  eventRegistry.registerPath({
+    method: 'put',
+    path: '/events/{id}',
+    tags: ['Event'],
+    request: {
+      params: CreateEventSchema,
+    },
+    responses: createApiResponse(EventSchema, 'Success'),
+  });
+
+  router.put('/:id', async (req: Request, res: Response) => {
+    try {
+      const id = req.params.id as string;
+      const parsedEvent = UpdateEventSchema.parse(req);
+      const serviceResponse = await eventService.update(id, parsedEvent);
+      handleServiceResponse(serviceResponse, res);
+    } catch (error) {
+      if (error instanceof ZodError) {
+        const err = new ServiceResponse(ResponseStatus.Failed, error.message, null, StatusCodes.BAD_REQUEST);
+        return handleServiceResponse(err, res);
+      }
+    }
+  });
+
+  eventRegistry.registerPath({
+    method: 'delete',
+    path: '/events/{id}',
+    tags: ['Event'],
+    responses: createApiResponse(z.object({}), 'Success'),
+  });
+
+  router.delete('/:id', validateRequest(GetEventSchema), async (req: Request, res: Response) => {
+    const id = req.params.id as string;
```

Comment:
We don't need as string, it is already inferred as a string because all route parameters in req.params are strings by default.